packages/mermaid/src/diagrams/state/stateRenderer-v2.js

```diff
@@ -324,7 +327,7 @@ const setupDoc = (g, parentParsedItem, doc, diagramStates, diagramDb, altFlag) =
           setupNode(g, parentParsedItem, item.state1, diagramStates, diagramDb, altFlag);
           setupNode(g, parentParsedItem, item.state2, diagramStates, diagramDb, altFlag);
           const edgeData = {
-            id: 'edge' + graphItemCount,
+            id: 'edge' + graphItemCount + `_${item.state1.id}_${item.state2.id}`,
```

Comment:
<!-- Questions are appropriate if you have a potential concern but are not quite sure if it's relevant or not. Asking the author for clarification or investigation can lead to a quick resolution.  -->
**question:** ‏maybe we could follow flowchart edge id standard?  
<img width="519" alt="Captura de Tela 2024-05-04 às 09 28 41" src="https://github.com/mermaid-js/mermaid/assets/54173994/3949726c-1fbd-41ab-ab0c-07c3a7be074f">

Reply:
That should be fine. 
Let's create a function to generate ID for edges, and use it everywhere, so it's consistent. 

Also, string templates are preferred over concatenation. 
```suggestion
            id: `edge_${graphItemCount}_${item.state1.id}_${item.state2.id}`,
```

Reply:
@sidharthv96 If I understood correctly, are we going to create a function for the state diagram? Or will it be something global to be used in the other diagrams?

Reply:
Something in the common utils seems like the way to go, as the plan is to have a consistent pattern across diagram types. 

I'm not really sure if we need the prefix and suffix though. 

```ts
export const getEdgeID = (
  from: string,
  to: string,
  {
    counter = 0,
    prefix = 'edge',
    suffix
  }: { counter?: number; prefix?: string; suffix?: string } = {}
) => {
  return `${prefix}_${from}_${to}_${counter}${suffix ? `_${suffix}` : ''}`;
};
```

Reply:
@sidharthv96 I also think that the prefix/suffix is not necessary, but I'm not sure if there's any code today that depends on it.

Looking at some diagrams:
- Dataflow = L-$FROM-$TO-$COUNTER
- Classchart = id_$FROM_$TO_$COUNTER
- Flowchart = L-$FROM-$TO-$COUNTER

The other diagrams do not have IDs in their respective relationships, honestly looking at the codebase I don't think it's necessary but I don't know if people usually use it haha.
But we can keep it to avoid any breaking changes.